src/lang.rs

```diff
@@ -446,6 +451,58 @@ named!(limit<Span, Positioned<InlineOperator>>, with_pos!(ws!(do_parse!(
     })
 ))));
 
+/// HACK: work-around to get around the incompatibility of verify! with non-Copy objects
+/// This is solved in nom 5.0 (https://github.com/Geal/nom/issues/510)
+/// TODO: remove after updating to nom 5.0
+macro_rules! verify_ref {
+  // Internal parser, do not use directly
+  (__impl $i:expr, $submac:ident!( $($args:tt)* ), $submac2:ident!( $($args2:tt)* )) => (
+    {
+      use nom::lib::std::result::Result::*;
+      use nom::{Err,ErrorKind};
+
+      let i_ = $i.clone();
+      match $submac!(i_, $($args)*) {
+        Err(e)     => Err(e),
+        Ok((i, o)) => if $submac2!(&o, $($args2)*) {
+          Ok((i, o))
+        } else {
+          Err(Err::Error(error_position!($i, ErrorKind::Verify)))
+        }
+      }
+    }
+  );
+  ($i:expr, $submac:ident!( $($args:tt)* ), $g:expr) => (
+    verify_ref!(__impl $i, $submac!($($args)*), call!($g));
+  );
+  ($i:expr, $submac:ident!( $($args:tt)* ), $submac2:ident!( $($args2:tt)* )) => (
+    verify_ref!(__impl $i, $submac!($($args)*), $submac2!($($args2)*));
+  );
+  ($i:expr, $f:expr, $g:expr) => (
+    verify_ref!(__impl $i, call!($f), call!($g));
+  );
+  ($i:expr, $f:expr, $submac:ident!( $($args:tt)* )) => (
+    verify_ref!(__impl $i, call!($f), $submac!($($args)*));
+  );
+}
+
+named!(split<Span, Positioned<InlineOperator>>, with_pos!(ws!(do_parse!(
+    tag!("split") >>
+    from_column_opt: opt!(ws!(verify_ref!(
```

Comment:
We should probably structure the syntax so that we don't need this `verify`. Lets make it `from xyz` eg.

`split on "," from inp as split_row` 

This is also nice because it matches the `json` operator.

Reply:
Yeah - I had it as `from` originally. The reason I thinking about this syntax was because of this case:

Input:
```
INFO darren,100,hello
```

Agrind:
```
* | parse "* *" as level, row | split on "," from row
```

Originally, the output was:
```
[level=INFO row="darren,100,hello" _split=[darren, 100, hello]]
```

But I wanted the split operator to overwrite the column if an input column is specified so it would work like:

Output:
```
[level=INFO row=[darren, 100, hello]]
```

Then the syntax reads more naturally to me like:
```
* | parse "* *" as level, row | split row on ","
```

Instead of:
```
* | parse "* *" as level, row | split on "," from row
```

But I also see that it would be nice to match the `from [input_column]` syntax like json.

Reply:
Yeah I see the appeal, but since none of the other operators have that overwrite behavior, I think we may want to require a slightly more verbose construction to get that to happen:
`split on "," from row as row` if they actually want to overwrite an existing key